server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -1163,15 +1163,15 @@ public TestClusterNode currentMaster(ClusterState state) {
             TestClusterNode(DiscoveryNode node) throws IOException {
                 this.node = node;
                 final Environment environment = createEnvironment(node.getName());
-                masterService = new FakeThreadPoolMasterService(node.getName(), "test", deterministicTaskQueue::scheduleNow);
+                threadPool = deterministicTaskQueue.getThreadPool(runnable -> CoordinatorTests.onNodeLog(node, runnable));
```

Comment:
Many of the rest of the changes, like this one, are to ensure that we use the same `ThreadPool` instance for entering system context in the master service and then for asserting that publications are sent in system context. Before this change, we were creating multiple threadpool instances which was ok since we were ignoring their stateful behaviour.